src/screens/OtherProfile/hooks/useOtherProfileScreenHooks.tsx

```diff
@@ -0,0 +1,158 @@
+  };
+
+  const fetchFeeds = async (offset = 0) => {
+    setIsLoading(true);
+    try {
+      const result = await getOtherFeedsInProfile(targetUserProfileId, offset);
+      const {data: feedOtherProfile} = result;
+      const mapNewData = mappingColorFeed(feedOtherProfile);
+      if (offset === 0) {
+        setOtherProfileFeed(mapNewData, dispatchOtherProfile);
+        StorageUtils.otherProfileFeed.setForKey(targetUserProfileId, JSON.stringify(mapNewData));
+      } else {
+        const clonedFeeds = [...feeds, ...mapNewData];
+        setOtherProfileFeed(clonedFeeds, dispatchOtherProfile);
+        StorageUtils.otherProfileFeed.setForKey(targetUserProfileId, JSON.stringify(clonedFeeds));
+      }
+      setPostOffset(Number(result.offset));
+    } catch (e) {
+      console.log('error', e);
+    }
+    setIsLoading(false);
+  };
+
+  const fetchOtherProfileData = async () => {
+    try {
+      const result = await getOtherProfile(username);
+      if (result.code === 200) {
+        const data = result?.data;
+        setOtherProfileData(data);
+        setOtherProfileUserId(data.user_id);
+
+        StorageUtils.otherProfileData.setForKey(targetUserProfileId, JSON.stringify(data));
+      }
+    } catch (e) {
+      console.log('error fetching other profile data', e);
+      if (e?.response?.data?.message) {
+        SimpleToast.show(e.response.data.message, SimpleToast.SHORT);
+      }
+    }
+  };
+
+  const fetchBlockStatus = async (withFeedsFetching = false) => {
+    const sendData = {
+      user_id: targetUserProfileId
+    };
+    try {
+      const processGetBlock = await checkUserBlock(sendData);
+      if (processGetBlock.status === 200) {
+        const isBlockedFromApi = processGetBlock.data.data.blocked;
+        const isBlockingFromApi = processGetBlock.data.data.blocker;
+
+        setIsBlocked(isBlockedFromApi);
+        setIsBlocking(isBlockingFromApi);
+
+        StorageUtils.blockedStatus.setForKey(targetUserProfileId, isBlockedFromApi.toString());
+        StorageUtils.blockingStatus.setForKey(targetUserProfileId, isBlockingFromApi.toString());
+
+        if (withFeedsFetching) await fetchFeeds();
+      }
+    } catch (e) {
+      console.log('error fetching block status', e);
+    }
+  };
+
+  const initData = async () => {
+    await getCache();
+
+    await Promise.all([fetchBlockStatus(true), fetchOtherProfileData(), fetchFeeds()]);
+
+    setIsProfileFetching(false);
+    setIsLoading(false);
+  };
+
+  React.useEffect(() => {
+    initData();
+
+    return () => {
+      setOtherProfileFeed([], dispatchOtherProfile);
+    };
+  }, []);
+
+  return {
+    feeds,
+    otherProfileData,
+    isBlocked,
+    isBlocking,
+    isLoading,
+    isProfileFetching,
+    postOffset,
+
+    refetchFeeds: fetchFeeds,
+    refetchOtherProfile: () => fetchOtherProfileData(),
+    refetchBlockStatus: () => fetchBlockStatus(false),
+
+    setOtherProfileData
+  };
+};
+
+export default useOtherProfileScreenHooks;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code looks good overall, but there are a few areas that could be improved for better error handling and maintainability.



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
Closed